File: MediaLoader/src/main/java/com/app/medialoader/tinyhttpd/codec/ResponseEncoder.java
package com.app.medialoader.tinyhttpd.codec;

import com.app.medialoader.tinyhttpd.response.Response;

import java.io.IOException;

/**
 * {@link Response}的编码器
 *
 * @author vincanyang
 */
public interface ResponseEncoder<T extends Response> {

    byte[] encode(T response) throws IOException;
}

File: MediaLoader/src/main/java/com/app/medialoader/tinyhttpd/codec/ResponseEncoder.kt
package com.app.medialoader.tinyhttpd.codec

import com.app.medialoader.tinyhttpd.response.Response
import java.io.IOException

/**
 * [Response]的编码器
 *
 * @author vincanyang
 */
interface ResponseEncoder<T : Any> {
    @Throws(IOException::class)
    fun encode(response: Any): ByteArray
}